lib/backends/sdc/volumes.js

```diff
@@ -222,7 +244,17 @@ function deleteVolume(params, options, callback) {
                 }
 
                 pollVolumeState(volumeUuid, volapiClient, ['deleted', 'failed'],
-                    next);
+                    function _onPollComplete(err, vol) {
+                        var deleteErr = err;
+
+                        if (!err && vol.state === 'failed') {
```

Comment:
Check that `vol` is not undefined with `if (!err && vol && vol.state === 'failed')`, or maybe assert that `vol` is not undefined when `err` is falsy.